docs/machine-learning/tutorials/iris-clustering.md

```diff
@@ -0,0 +1,207 @@
+
+[!code-csharp[Add usings for paths](../../../samples/machine-learning/tutorials/IrisClustering/Program.cs#2)]
+
+## Create a learning pipeline
+
+Add the following additional `using` directives to the top of the *Program.cs* file:
+
+[!code-csharp[Add Microsoft.ML usings](../../../samples/machine-learning/tutorials/IrisClustering/Program.cs#3)]
+
+In the `Main` method, replace the `Console.WriteLine("Hello World!")` with the following code:
+
+[!code-csharp[Call the Train method](../../../samples/machine-learning/tutorials/IrisClustering/Program.cs#4)]
+
+The `Train` method trains the model. Create that method just below the `Main` method, using the following code:
+
+```csharp
+private static PredictionModel<IrisData, ClusterPrediction> Train()
+{
+
+}
+```
+
+The learning pipeline loads all of the data and algorithms necessary to train the model. Add the following code into the `Train` method:
+
+[!code-csharp[Initialize pipeline](../../../samples/machine-learning/tutorials/IrisClustering/Program.cs#5)]
+
+## Load and transform data
+
+The first step to perform is to load the training data set. In our case, training data set is stored in the text file with a path defined by the `_dataPath` field. Columns in the file are separated by the comma (","). Add the following code into the `Train` method:
+
+[!code-csharp[Add step to load data](../../../samples/machine-learning/tutorials/IrisClustering/Program.cs#6)]
+
+The next step is to combine all of the feature columns into the **Features** column using the <xref:Microsoft.ML.Transforms.ColumnConcatenator> transformation class. By default, a learning algorithm processes only features from the **Features** column. Add the following code:
+
+[!code-csharp[Add step to concatenate columns](../../../samples/machine-learning/tutorials/IrisClustering/Program.cs#7)]
+
+## Choose a learning algorithm
+
+After adding the data to the pipeline and transforming it into the correct input format, you select a learning algorithm (**learner**). The learner trains the model. ML.NET provides a <xref:Microsoft.ML.Trainers.KMeansPlusPlusClusterer> learner that implements [k-means algorithm](https://en.wikipedia.org/wiki/K-means_clustering) with improved method for choosing the initial cluster centroids.
+
+Add the following code into the `Train` method following the data processing code added in the preceding section:
+
+[!code-csharp[Add a learner step](../../../samples/machine-learning/tutorials/IrisClustering/Program.cs#8)]
+
+Note that we specify that the data set should be split in three clusters.
+
+## Train the model
+
+The steps added in the preceding sections prepared the pipeline for training. None of the steps has been executed. The `pipeline.Train<TInput, TOutput>` method produces the model that takes in an instance of the `TInput` type and outputs an instance of the `TOutput` type. Add the following code into the `Train` method:
```

Comment:
The steps added in the preceding sections prepared the pipeline for training, however, none have been executed.